docs/building.md

```diff
@@ -1,10 +1,16 @@
 ## Building from source
 
-Ensure you have JDK 21 (or newer) and Git installed
+Ensure you have **JDK 21** and Git installed
 
     java -version
     git --version
 
+Newer versions of the JDK are not supported. If you have multiple JDK versions
+installed, you can specify which one to use during the build by setting the `JAVA_HOME`
+environment variable (this should be the directory containing `/bin/` or `/jre/`).
+
+    JAVA_HOME=/path/to/jdk-21/ ./mvnw clean install
+
```

Comment:
Not sure we need to add these instructions to be honest. AFAIK most folks will be using a version manager such as [SDKMAN!](https://sdkman.io/), which can handle this for them.

Reply:
Until we have an `.sdkmanrc` file in our repo, I think I would go with this approach.

Reply:
Just a note: I'm not an every-day Java developer so I don't know/have all of the fancy tools ingrained to make development easy. I'm a drive-by user coming to this project because we were encountering a problem and I wanted to add some more debug logs in the code to see if it was a Keycloak oddity or something spec'ed.

In the future, if it's expected for people to use a tool like [SDKMAN!](https://sdkman.io/), it should be in the docs and not assumed. Luckily, multiple versions of Java/JDK works pretty easy in my case without having to figure this piece out as well. Suggesting the tool in the docs and having the version automatically picked up from config in the repo sounds pretty great though (it just works) :+1:. Plus, it's less system-level installs necessary.